File: prototype/c-sharp/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Prototype
{
    // Employee sınıfına klonlama yeteneği kazandırmak için tanımladığımız arayüzü uyguladık.
    class Employee : IClonablePrototype<Employee>
    {
        string firstName;
        string lastName;

        public Employee(string firstName, string lastName)
        {
            this.firstName = firstName;
            this.lastName = lastName;
        }

        public Employee Clone()
        {
            return base.MemberwiseClone() as Employee;
        }
    }
}

File: prototype/java/Employee.java
package Prototype;

// Employee sınıfına klonlama yeteneği kazandırmak için tanımladığımız arayüzü uyguladık.
public class Employee implements IClonablePrototype<Employee> {
    String firstName;
    String lastName;

    public Employee(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    @Override
    public Employee clone() throws CloneNotSupportedException {
        return (Employee) super.clone();
    }
}
